features/CTX000_Presentation-colours-and-textures.feature

```diff
@@ -0,0 +1,22 @@
+@implementer-agreement
+@CTX
+@version1
+@E00020
+
+Feature: CTX000 - Presentation colours and textures
+    The rule verifies the presence of colour, texture and other presentation appearance information to objects.
+    https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/concepts/Product_Shape/Product_Geometry_Colour/content.html
+
+    Scenario Outline: Check for activation
+
+        Given an <entity type>
+        Given its attribute <attribute> 
+
+        Then The IFC model contains information on the selected functional part
+
+        Examples:
+            | entity type | attribute |
+            | IfcSolidModel | StyledByItem |
+            | IfcTessellatedFaceSet | StyledByItem |
+            | IfcTessellatedFaceSet | HasColours |
+
```

Comment:
There's another scenario missing for styled materials. One of the most ridiculous things in the IFC spec in terms of complexity. Keep in mind the `...` because you can have multiple types of materials e.g a material layer set, but ultimately ending up always at a Material. So we can use the traversal feature we built earlier.

IfcProduct <- IfcRelAssociatesMaterial -> ... -> Material <- IfcMaterialDefinitionRepresentation -> IfcStyledRepresentation -> IfcStyledItem

Reply:
You mean what we did at the statement Given all referenced instances? 
```
@gherkin_ifc.step("All referenced instances")
def step_impl(context, inst):
    # Note that this includes `inst` as the first element in this list
    instances = context.model.traverse(inst)
    yield ValidationOutcome(instance_id=instances, severity=OutcomeSeverity.PASSED)
```

So would this be something like
```
Given an IfcProduct 
Given its attribute HasAssociations 
Given All referenced instances
Given its attribute HasRepresentation 
Given its attribute Representations
Given its attribute Items
Given its entity type is 'IfcStyledItem'
```
?

Reply:
Thanks. Great work.

Does this work in IFC4+ as well? Since the HasAssociations moved around to IfcObjectDefinition.

![afbeelding](https://github.com/user-attachments/assets/d68ecf6e-96e3-434b-a004-a38488710c99)

Reply:
I think so. The [testfile](https://github.com/buildingSMART/ifc-gherkin-rules/pull/312/commits/948bdeefc3407c0d9d9ce0c45382578cbbe8cff2#diff-e460442251d5a59af6dcf9b35f4cb74ee1fa7a5ecdd1b082263ae5e887d93adf) for this scenario is IFC4X3_ADD2

```
>>> inst
#2814=IfcRailType('0nXz8k3L16HPW1otmrVMMM',#1,'Rail',$,$,$,$,$,$,.NOTDEFINED.)
>>> inst.HasAssociations
(#2816=IfcRelAssociatesMaterial('21AGKYInDEAexMZP3z_WRc',#1,'Type Material',$,(#2814),#18),)
>>> mat = [i for i in f.traverse(inst.HasAssociations[0]) if i.is_a('IfcMaterial')][0]
>>> mat
#18=IfcMaterial('Rail Material',$,$)
>>> mat.HasRepresentation[0].Representations[0].Items[0].is_a()
'IfcStyledItem'
```